client/web/compose/src/views/Admin/Pages/Builder.vue

```diff
@@ -570,20 +595,38 @@ export default {
     },
 
     handleClone () {
+      const { namespaceID = NoID } = this.namespace
       let page = this.page.clone()
-      page = {
-        ...page,
-        pageID: NoID,
-        title: this.$t('copyOf', { title: this.page.title }),
-        handle: '',
-      }
 
-      const { namespaceID = NoID } = this.namespace
-      this.createPage({ namespaceID, ...page })
-        .then(({ pageID }) => {
-          this.$router.push({ name: 'admin.pages.builder', params: { pageID } })
+      this.loadPages({ namespaceID, force: true })
+        .then(pages => {
+          const pageTitles = pages.map(p => new compose.Page(p).title)
+          let newPageTitle = this.$t('copyOf', { title: this.page.title })
+          const countDuplicates = this.countDuplicatePages(pageTitles)
+
+          if (countDuplicates === 2) {
+            newPageTitle = this.page.title + ' (' + countDuplicates + ')'
+          } else if (countDuplicates > 2) {
+            newPageTitle = this.page.title.includes(this.$t('copyOf'))
+              ? this.page.title.substr(0, this.page.title.lastIndexOf('(')) + '(' + countDuplicates + ')'
+              : this.$t('copyOf') + this.page.title + ' (' + countDuplicates + ')'
+          }
+
+          page = {
+            ...page,
+            pageID: NoID,
+            title: newPageTitle,
+            handle: '',
+          }
+
+          this.createPage({ namespaceID, ...page })
+            .then(page => {
+              this.page = new compose.Page(page)
```

Comment:
I dont think we need to set this.page, since you get redirected so its just confusing